src/shared/components/query/QueryStore.ts

```diff
@@ -297,8 +230,28 @@ export class QueryStore
 	{
 		let ids:string[] = this._selectedStudyIds.keys();
 		const selectableStudies = this.selectableStudiesSet;
-		ids = ids.filter(id=>!!selectableStudies[id]);
-		return this.forDownloadTab ? ids.slice(-1) : ids;
+		let selectableIds = ids.filter(id=>!!selectableStudies[id]);
+
+		if(ids.length !== selectableIds.length) {
+			let entities = this.caseIds.trim().split(/\s+/g);
+			let normStudies:string [] = entities
+			.map(entity=>{
+				let splitEntity = entity.split(':');
+				if (splitEntity.length === 2) {
+					const study = splitEntity[0];
+					const id = splitEntity[1];
+					return study;
+				}
+			})
+			.filter(element => element !== undefined) as string[];
+
+			normStudies = _.uniq(normStudies);
+
+			selectableIds = normStudies.filter(id=>!!selectableStudies[id]);

```

Comment:
same thing here.  this can be accomplished inside reduce and without extra iteration